File: JavaProject/GrowNet/src/main/java/ai/nektron/grownet/InhibitoryNeuron.java
package ai.nektron.grownet;

public class InhibitoryNeuron extends Neuron {
    public InhibitoryNeuron(String id, LateralBus bus, SlotConfig cfg, int slotLimit) {
        super(id, bus, cfg, slotLimit);
    }
    @Override protected void fire(double inputValue) {
        bus.setInhibition(0.7); // tune later
    }
}

File: src/cpp/InhibitoryNeuron.cpp
#include "InhibitoryNeuron.h"

namespace grownet {

    void InhibitoryNeuron::fire(double /*inputValue*/) {
        // One-tick attenuation; reset occurs in LateralBus::decay()
        if (bus != nullptr) {
            bus->setInhibitionFactor(0.7); // gamma; tune later
        }
    }

} // namespace grownet
